### PR TITLE
Add xlnt port

### PR DESCRIPTION
**Pull request by tfussell**
xlnt is a C++ library I wrote for reading and writing XLSX files. I would like to add support for installing xlnt using vcpkg with this pull request. Please let me know if there's anything I need to do to improve this port. (Closes tfussell/xlnt#85).

### CODE REVIEW

**Comment by codicodi** [comment]
I think it would be best if dependencies of your library would be provided by vcpkg. Unfortunately we don't have ports of libstudxml and botan, so IMHO they should be added first.
It would be also nice if your port automatically inserted `XLNT_STATIC `macro to your headers after build.

As a sidenote not related to port: I noticed that constructor for ` document_security` class isn't defined anywhere - shouldn't it be defaulted?

**Comment by tfussell** [comment]
Thanks for the feedback @codicodi. It would be nice to use vcpkg for all of the dependencies, but I'm targeting Windows, OSX, and Linux so I think I'll have to stick with Git submodules for the time being. I'm not sure how other vcpkg ports have dealt with this.

You're right about the XLNT_STATIC definition. I'll see if I can get that working.

As for document_security, defaulting the constructor is a good idea. Xlnt does have some unused methods missing definitions since my goal was to implement the core functionality and then let the community provide any advanced fucntionality they might need. The spec is just too big for me to get everything done on version 1.0.

**Comment by ras0219-msft** [comment]
We do need to use system dependencies when they are available, to enable

- Upgrading/security patching
- Static linking (to avoid symbol conflicts with the system copies)
- Sane debugging (`#include <expat>` does the same thing everywhere)

For now, this means cryptopp, expat, zlib, and utfcpp.

Looking at [1], I think the least intrusive way to do this would be to add a non-default cmake macro that would disable the `xlnt.third-party` target and instead run a series of `find_library` calls that grab the appropriate prebuilt versions and assemble them into a cmake pseudo-target.

[1] https://github.com/tfussell/xlnt/blob/master/third-party/CMakeLists.txt

**Comment by tfussell** [comment]
Thanks for taking a look at my port @ras0219-msft. It does make sense to use system libraries when possible. I was simply trying to reduce friction when using my library by building all of the third-party libraries directly. I've nearly sorted out how to use find_package to locate vcpkg libraries. I'll get the port modified and update the pull request accordingly soon.

**Comment by ras0219-msft** [comment]
@tfussell Any progress here?

**Comment by tfussell** [comment]
I'm working hard to get this working. This cross-platform support is turning out to be complicated. I hope to have an update for you by the weekend. Sorry it's taking so long. @ras0219-msft

**Comment by ras0219-msft** [comment]
No problem at all, I just wanted to check in.

Please take your time to make sure it's in a state you can maintain going forward. I'd be happy to help on any decisions you're unsure about.

File: ports/xlnt/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,49 @@
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+    set(STATIC "OFF")
+else()
+    set(STATIC "ON")
+endif()
+
+include(vcpkg_common_functions)
+find_program(GIT git)
+
+set(GIT_URL "https://github.com/tfussell/xlnt.git")
```
Comment (Mixaill) [inline ports/xlnt/portfile.cmake]: Maybe we should switch to download revision by URL?

`https://github.com/<user>/<project>/archive/<hash>.zip`

`https://github.com/tfussell/xlnt/archive/f908dca6fe342f44b814073947aaccc69df7c9b6.zip`

**Review by Mixaill — CHANGES_REQUESTED** [review]

**Comment by tfussell** [comment]
@Mixaill Thanks for the suggestion. I've reconfigured the xlnt portfile based on your recommendation and some other example ports I found. @ras0219-msft It took more work than I anticipated, but I believe my library and associated port now meet all of your requirements. static/dynamic and x86/x64 builds are all working on my end. Let me know if you need me to do anything else for this to be merged.

**Comment by ras0219-msft** [comment]
Looks great, thanks for the hard work!

**Pull request merged**